tracing-subscriber/src/registry/sharded.rs

```diff
@@ -198,13 +202,16 @@ impl Subscriber for Registry {
     fn event(&self, _: &Event<'_>) {}
 
     fn enter(&self, id: &span::Id) {
-        CURRENT_SPANS.with(|spans| {
-            spans.borrow_mut().push(self.clone_span(id));
-        })
+        let spans = self.current_spans.get_or_default();
+        spans.borrow_mut().push(self.clone_span(id));
```

Comment:
nit, take it or leave it: could just write:
```suggestion
        self.current_spans
            .get_or_default()
            .borrow_mut()
            .push(self.clone_span(id));
```
but it's not a blocker.